Dockerfile

```diff
@@ -59,30 +63,91 @@ RUN --mount=type=cache,target=/opt/ccache \
     CMAKE_PREFIX_PATH="$(dirname $(which conda))/../" \
     python setup.py install
 
-FROM conda as conda-installs
-ARG PYTHON_VERSION=3.11
-ARG CUDA_PATH=cu121
-ARG CUDA_CHANNEL=nvidia
+# Stage 5: Torch dependencies Installation
+FROM conda as torch-packages
 ARG INSTALL_CHANNEL=whl/nightly
-# Automatically set by buildx
-RUN /opt/conda/bin/conda update -y -n base -c defaults conda
-RUN /opt/conda/bin/conda install -y python=${PYTHON_VERSION}
-
+ARG CUDA_PATH=cu121
+ARG PYTORCH_VERSION=2.7.0.dev20250217 #we need to set this 
```

Comment:
@altman We need to choice a default here and set it externally when we have nighlty/test channel.